src/aleph/vm/orchestrator/status.py

```diff
@@ -119,8 +119,15 @@ async def check_internet(session: ClientSession, vm_id: ItemHash) -> bool:
     """Check that the VM has internet connectivity. This requires DNS, IP, HTTP and TLS to work."""
     try:
         result: dict = await get_json_from_vm(session, vm_id, "/internet")
-        assert result["result"] == HTTPOk.status_code
+
+        # The HTTP Header "Server" must always be present in the result.
         assert "Server" in result["headers"]
+
+        # The diagnostic VM returns HTTP 200 with {"result": False} when cannot connect to the internet.
+        # else it forwards the return code if its own test endpoint.
+        if result.get("result") != HTTPOk.status_code:
+            return False
```

Comment:
Please, can you add an automatic test to avoid having that issue in the future?

Reply:
Absolutely, thanks for reminding me. This is a blocker.

Reply:
Tests added.